velox/docs/develop/debugging/tracing.rst

```diff
@@ -0,0 +1,146 @@
+=======
+Tracing
+=======
+
+
+Background
+----------
+
+The query trace tool helps analyze and debug query performance and correctness
+issues. It helps prevent interference from the test noises in a production
+environment (such as storage, network etc) by allowing replay of a part of the
+query plan and data set in an isolated environment such as a local machine.
+This is much more efficient for query performance debugging as we don't have to
+replay the whole query in a production environment.
+
+How Tracing Tool Works
+----------------------
+
+The tracing process consists of two phases: tracing and replaying.
+
+**Tracing Phase**
+
+- When the query starts, the task records the metadata including query plan fragment,
+  query configuration, and connector properties.
+- During the query running, each traced operator records the input vectors and saves
+  in the specified storage location.
+- The metadata are serialized using json format and operator data inputs are serialized
+  using `Presto serializer <https://prestodb.io/docs/current/develop/serialized-page.html>`_.
+
+**Replaying Phase**
+
+- Read and deserialize the recorded query plan in tracing phase, extract the target plan
+  node, and assemble a plan fragment with a customized source and sink nodes. The source
+  node reads the input from the serialized operator inputs on storage and sink operator
+  prints or logs out the execution stats.
+- Build a query/task with the assembled plan fragment in step 1.
```

Comment:
query/task -> task

step 1 - since steps are not numbered, that reference is confusing

Reply:
Fixed